File: backend/CoreLayer/models/Order/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CoreLayer.models.Order
{
	public class Order:BaseClass
	{
		//--Create class Order : BaseClass[ClientEmail - OrderDate - OrderState - ShippingAddress - deliveryMethod - OrderItems - SubTotal - GetTotal - PaymentIntentId]
		//--Create enum OrderState[Pending-PaymentReceived-PaymentFailed]
		//--Create class ShippingAddress[FirstName-LastName-Street-City-Country]
		//--Create class DeliveryMethod : BaseClass[ShortName - Description - Cost - DeliveryTime]
		//--Create class OrderItem : BaseClass[ProductItemOrdered - Price - Quantity] 
		//--Create class ProductItemOrdered Contain[ProductId, ProductName, PictureUrl
		public string ClientEmail { get; set; } = null!;
		public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.UtcNow;
		public OrderState orderState { get; set; } = OrderState.Pending;
		public ShippingAddress shippingAddress { get; set; } = null!;
		public DeliveryMethod? deliveryMethod { get; set; }
		public ICollection<OrderItem> orderItems { get; set; } = new HashSet<OrderItem>();
		public decimal SubTotal { get; set; }


		[NotMapped]
		public decimal Total => SubTotal + deliveryMethod?.Cost ?? 0;
		//OR
		//public decimal GetTotal()=> SubTotal + deliveryMethod?.Cost ?? 0;

		public string PaymentIntentId { get; set; } = string.Empty; // علشان لو مبعتش قيمة ميضربش اكسيبشن

	}
}

File: frontend_Angular_20/src/app/shared/models/order.ts
import { IAddress } from "./address";
/*
///////// ده الرد اللي هييجي من ال api 
//////// ف لازم عناصر اللي هنا تبقي زيه ب الظبط !
[
  {
    "id": 0,
    "clientEmail": "string",
    "orderDate": "2025-06-19T18:39:51.512Z",
    "orderState": "string",
    "shippingAddress": {
      "firstName": "string",
      "lastName": "string",
      "street": "string",
      "city": "string",
      "country": "string"
    },
    "deliveryMethodCost": 0,
    "items": [
      {
        "productId": 0,
        "productName": "string",
        "pictureUrl": "string",
        "price": 0,
        "quantity": 0
      }
    ],
    "subTotal": 0,
    "total": 0
  }
]

*/ 
export interface IOrderToCreate {
    basketId: string;
    //deliveryMethodId: number;
    shippingAddress: IAddress;
}

export interface IOrder {
    id: number;
    clientEmail: string;
    orderDate: string;
    shippingAddress: IAddress;
    deliveryMethodCost: number;
    items: IOrderItem[];
    subTotal: number;
    orderState: string;
    total: number;
  }
  
  export interface IOrderItem {
    productId: number;
    productName: string;
    pictureUrl: string;
    price: number;
    quantity: number;
  }